lib/internal/process/promises.js

```diff
@@ -144,20 +144,15 @@ function emitWarning(uid, reason) {
   } catch {}
 
   process.emitWarning(warning);
-  emitDeprecationWarning();
 }
 
 let deprecationWarned = false;
 function emitDeprecationWarning() {
-  if (unhandledRejectionsMode === kDefaultUnhandledRejections &&
```

Comment:
Won't this emit the warning every time rather than just once?

Also - doesn't this break the flag?

cc @BridgeAR

Reply:
The only call site of `emitDeprecationWarning()` is now L200 which is already guarded with `deprecationWarned `.

Also the only branch on `unhandledRejectionsMode` is on L180 so you can see what actions are done for each mode by simply looking at that switch.